### ISSUE TITLE
MASE metric not implemented per definition

### ISSUE THREAD

**Issue by arangatang** [awslabs/gluon-ts#1135]
## Description
The current implementation of the MASE metric calculation is not true to the definition of the metric. Our implementation is discontinuous and plain wrong in some scenarios. 

For example when forecasting on the constant dataset using some estimator, the current implementation reports a MASE of 0 which is impossible as the naive seasonal forecaster is optimal in this scenario. 

## To Reproduce
(Please provide minimal example of code snippet that reproduces the error. For existing examples, please provide link.)

```python
from gluonts.dataset.repository.datasets import get_dataset
from gluonts.evaluation.backtest import backtest_metrics
from gluonts.model.deepar import DeepAREstimator
from gluonts.mx.trainer import Trainer

ds = get_dataset("constant")
estimator = DeepAREstimator(
    freq=ds.metadata.freq,
    prediction_length=ds.metadata.prediction_length,
    trainer=Trainer(epochs=1),
)
pred = estimator.train(ds.train)
agg, _ = backtest_metrics(ds.test, pred)

print(agg['MASE'])
```

## Error message or code output

```
0.0 # it should be more than or equal to 1
```

**Comment by arangatang** [comment]
I suggest we add two versions: 

1. one which is the MASE metric as per its definition, thus allowing MASE to reach Infinity. 
2. one which is a continuous version of what we have

**Comment by arangatang** [comment]
Closing this as #1409 addresses this.
